server/functions/framework/compiled_function.go

```diff
@@ -198,10 +200,16 @@ func (c *CompiledFunction) Eval(ctx *sql.Context, row sql.Row) (interface{}, err
 	if err != nil {
 		return nil, err
 	}
-	// Convert the parameter values into their correct types
-	resultTypes := c.callableFunc.GetParameters()
 
-	if c.callableFunc.GetIsStrict() {
+	// Convert the parameter values into their target types
+	var targetParamTypes []pgtypes.DoltgresType
+	if c.varArgsType != pgtypes.DoltgresTypeBaseID_Invalid {
+		targetParamTypes = make([]pgtypes.DoltgresType, len(parameters))
+	} else {
+		targetParamTypes = c.callableFunc.GetParameters()
+	}
```

Comment:
We shouldn't special case this logic like this. We're adhering to the function resolution that Postgres does as _closely_ as possible. Any deviations that we make will only manifest as bugs later as queries get more complicated and general compatibility gets better. Our current testing is relatively non-existent, and the GMS engine tests won't catch any of these issues since they won't test Postgres function resolution.

Postgres doesn't have a concept of an "invalid" type. The closest is the `unknown` type, which has specific handling, and is a known quantity inserted during the parsing phase. We should remove this "invalid" type before it causes further deviations from Postgres' handling.